.github/workflows/kind-e2e-containerd.yaml

```diff
@@ -0,0 +1,164 @@
+        working-directory: ./
+        run: |
+          echo '::group:: install ko'
+          curl -L https://github.com/google/ko/releases/download/v0.9.3/ko_0.9.3_Linux_x86_64.tar.gz | tar xzf - ko
+          chmod +x ./ko
+          sudo mv ko /usr/local/bin
+          echo '::endgroup::'
+
+          echo "::group:: install gotestsum ${GOTESTSUM_VERSION}"
+          curl -L https://github.com/gotestyourself/gotestsum/releases/download/v${GOTESTSUM_VERSION}/gotestsum_${GOTESTSUM_VERSION}_linux_amd64.tar.gz | tar xzf - gotestsum
+          chmod +x ./gotestsum
+          sudo mv gotestsum /usr/local/bin
+          echo "::endgroup::"
+
+      - name: Check out code onto GOPATH
+        uses: actions/checkout@v2
+        with:
+          path: ./src/knative.dev/container-freezer
+
+      - name: Install KinD
+        run: |
+          set -x
+          # Disable swap otherwise memory enforcement doesn't work
+          # See: https://kubernetes.slack.com/archives/CEKK1KTN2/p1600009955324200
+          sudo swapoff -a
+          sudo rm -f /swapfile
+          curl -Lo ./kind https://github.com/kubernetes-sigs/kind/releases/download/${{ matrix.kind-version }}/kind-$(uname)-amd64
+          chmod +x ./kind
+          sudo mv kind /usr/local/bin
+
+      - name: Create KinD Cluster
+        run: |
+          set -x
+          # KinD configuration.
+          cat > kind.yaml <<EOF
+          apiVersion: kind.x-k8s.io/v1alpha4
+          kind: Cluster
+          nodes:
+          - role: control-plane
+            image: kindest/node:${{ matrix.k8s-version }}@${{ matrix.kind-image-sha }}
+          - role: worker
+            image: kindest/node:${{ matrix.k8s-version }}@${{ matrix.kind-image-sha }}
+          # This is needed in order to
+          # (1) support projected volumes with service account tokens. See
+          #     https://kubernetes.slack.com/archives/CEKK1KTN2/p1600268272383600
+          # (2) use a random cluster suffix
+          kubeadmConfigPatches:
+            - |
+              kind: ClusterConfiguration
+              metadata:
+                name: config
+              apiServer:
+                extraArgs:
+                  "service-account-issuer": "kubernetes.default.svc"
+                  "service-account-signing-key-file": "/etc/kubernetes/pki/sa.key"
+              networking:
+                dnsDomain: "${CLUSTER_SUFFIX}"
+          EOF
+          # Create a cluster!
+          kind create cluster --config kind.yaml
+
+      - name: Install Knative Serving and net-kourier
+        run: |
+          set -o pipefail
+          kubectl apply -f https://github.com/knative/serving/releases/download/knative-v${KN_SERVING_VERSION}/serving-crds.yaml
+          kubectl apply -f https://github.com/knative/serving/releases/download/knative-v${KN_SERVING_VERSION}/serving-core.yaml
+          kubectl apply -f https://github.com/knative/net-kourier/releases/download/knative-v${KN_SERVING_VERSION}/kourier.yaml
```

Comment:
Thinking out loud here... would it make sense to run this against the nightlies rather than a specific version? That would let us test against the latest serving code (more or less), without having to build from source...

Reply:
fixed
```shell
kubectl apply -f https://storage.googleapis.com/knative-nightly/serving/latest/serving-crds.yaml
kubectl apply -f https://storage.googleapis.com/knative-nightly/serving/latest/serving-core.yaml
kubectl apply -f https://storage.googleapis.com/knative-nightly/net-kourier/latest/kourier.yaml   
```

Reply:
Maybe give a matrix, which contains nightly version and newest release version(now v1.5)?

Reply:
It's a little bit overkill, Let's keep using nightly version.